kategory/src/main/kotlin/kategory/typeclasses/Monad.kt

```diff
@@ -72,5 +72,56 @@ fun <F, B> Monad<F>.binding(c: suspend MonadContinuation<F, *>.() -> HK<F, B>):
     return continuation.returnedMonad
 }
 
+@RestrictsSuspension
+open class StackSafeMonadContinuation<F, A>(val M: Monad<F>) : Serializable, Continuation<Free<F, A>> {
+
+    override val context = EmptyCoroutineContext
+
+    override fun resume(value: Free<F, A>) {
+        returnedMonad = value
+    }
+
+    override fun resumeWithException(exception: Throwable) {
+        throw exception
+    }
+
+    internal lateinit var returnedMonad: Free<F, A>
+
+    operator suspend fun <B> HK<F, B>.not(): B = this.bind()
+
+    suspend fun <B> HK<F, B>.bind(): B = bind { Free.liftF(this) }
+
+    suspend fun <B> Free<F, B>.bind(): B = bind { this }
```

Comment:
Unfortunately Kotlin does not allow for disambiguation via generic constrains choosing the most concrete type otherwise I could have done:
```kotlin
suspend fun <B, FK : FreeKind<F, B>> FK.bind(): B = bind { this.ev() }
```
Removing the Free version to keep only the one based on `HK` would not allow for recursive programs because recursive programs have to be typed and return `Free<F,A>` from the `binding` block as in https://github.com/kategory/kategory/blob/a2bdb99c48123bd92afef0b8eda871c1bc52348c/kategory/src/test/kotlin/kategory/laws/MonadLaws.kt#L85

Binding over `FreeKind<F,A>` would just give you back the same value you bound wrapped in Free and would fail at compile time which IMO is the best we can do now. If you have other ideas I can try something else in a separate PR to improve syntax in bindings.